src/toil/test/provisioners/cgcloud/provisionerTest.py

```diff
@@ -265,3 +266,44 @@ def testAutoScaledCluster(self):
     @integrative
     def testAutoScaledSpotCluster(self):
         self._test(autoScaled=True, spotInstances=True)
+
+
+restartScript = """\"from toil.job import Job
+import argparse
+import os
+
+
+def f0(job):
+    if 'FAIL' in os.environ:
+        raise RuntimeError('failed on purpose')
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    Job.Runner.addToilOptions(parser)
+    options = parser.parse_args()
+    i = Job.Runner.startToil(Job.wrapJobFn(f0), options)
+\""""
+
+
+class CGCloudRestartTest(AbstractCGCloudProvisionerTest):
+
+    def _getScript(self):
+        self.scriptName= "restartScript.py"
+        self._leader('echo %s > %s' % (restartScript, 'restartScript.py'), shell=True)
+
+    def _runScript(self, toilOptions):
+        # clean = onSuccess
+        restartCompatible = lambda x : ('clean' not in x and 'retryCount' not in x)
+        newOptions = [option for option in toilOptions if restartCompatible(option)]
```

Comment:
In this case isn't the generator expression just as expensive as the list comprehension? The result has to be evaluated before being unpacked, right? 

If they're equivalent I would rather stick with the list comprehension since it's a little more conventional and you can inspect the resulting 'newOptions' variable when debugging.